### DISCUSSION TITLE
Item-level cancellations, Returns and Replacements

### DISCUSSION THREAD

**Discussion by ravi-prakash-v** [beckn/protocol-specifications#323] [Core Specification]
I wanted to start a quick discussion on the topic of cancellations, returns and replacements of individual items in an Order. 

While an order-level cancellation can be achieved using the `cancel / on_cancel` API flow, item-level cancellations have been supported using the following approach in the upcoming version (1.x) of the specification. 

This is how the `Item` schema currently looks like
```
Item:
    ...
    cancellation_terms:
         ...
         return_policy:
               return_to:
               return_by:
               ...
    replacement_policy:
         replace_by:
         ...
```

The current [pre-release 1.x](https://github.com/beckn/protocol-specifications/tree/release-1.x) version bundles `ReturnTerm` inside the `CancellationTerm` schema. 

This is based on a logical assumption that _the return (without replacement) of any item in an Order is always a consequence of a cancellation of that item_. 

In case the buyer wants a _replacement_ of an item, there is a separate `ReplacementPolicy` schema that is present outside the `CancellationTerm` schema as an attribute of the `Item` schema.

One approach is to put `return_terms` should be outside the `CancellationTerms` schema, however I fail to see a single scenario where a return (without replacement) is possible without an item-level cancellation. Maybe I am missing something. 

Similarly, should `ReplacementTerm` be independent of `CancellationTerm` or an attribute of cancellations?

Simply put, two questions need to be answered.

1. Should `ReturnTerm` be independent or a property of the `CancellationTerm` schema?
2. Should `ReplacementTerm` be independent or a property of the `CancellationTerm` schema?

@pramodkvarma @venkatramanm @nitinmish @BLR-0118 @sjthnrk, your inputs would be helpful.

**Comment by sankarshanmukhopadhyay** [comment]
> In case the buyer wants a _replacement_ of an item, there is a separate `ReplacementPolicy` schema that is present outside the `CancellationTerm` schema as an attribute of the `Item` schema.> 

Can the buyer seek a replacement for an item? From their perspective, they desire a return and the seller can offer a refund or a replacement (based on the reason offered for a return)

> Simply put, two questions need to be answered.
> 
>     1. Should `ReturnTerm` be independent or a property of the `CancellationTerm` schema?
> 

I would think it to be independent. Additionally, should a buyer be able to cancel an order if the item has been received and set up for a return? It really means that the fulfilment was for a defective item and the fulfillment is through either a replacement or a refund (both of which are triggered by a return)

>     2. Should `ReplacementTerm` be independent or a property of the `CancellationTerm` schema?
